src/components/CreateSmartList/index.js

```diff
@@ -333,96 +290,103 @@ const CreateSmartList = () => {
     })
   }
 
-  const sortedGroceriesByDaysAverage = arrayOfItemsByAverage[0].sort(
-    (a, b) => b.daysAverage - a.daysAverage,
-  )
+  const sortdaysAverage = (array) =>
+    array.sort((a, b) => b.daysAverage - a.daysAverage)
 
-  const sortedProductsByDaysAverage = arrayOfItemsByAverage[1].sort(
-    (a, b) => b.daysAverage - a.daysAverage,
+  const sortedGroceriesByDaysAverage = sortdaysAverage(arrayOfItemsByAverage[0])
```

Comment:
ta tablica `arrayOfItemsByAverage` to koncepcyjnie trochę błąd - co jeżeli zamieni się gdzieś miejscami któryś z elementów, albo zniknie? Wtedy reszta się wysypie :P tablicy się używa tylko do elementów, które mają taką samą zawartość i logika działania (poza sortowaniem) nie zależy od ich kolejności. W takim przypadku lepszy będzie obiekt po kluczach, albo niezależne obiekty.

Tutaj by wystarczyło 
const sortedGroceriesByDaysAverage = sortdaysAverage(itemsListFromGroceriesWithAverage)

Reply:
Zmieniłem

Reply:
Poprawiłem